src/site/content/en/angular/setup-dynamic-server-side-rendering-with-angular/index.md

```diff
@@ -0,0 +1,143 @@
+---
+layout: post
+title: "Set up dynamic server-side rendering with Angular"
+subhead: |
+  Dynamically server-side rendering your pages per route using Angular Universal schematics.
+hero: angular-universal-motorcycle.jpeg
+alt: close-up of black and brown motorcycle dashboard
+date: 2019-10-01
+description: |
+  Learn how to set up dynamic server-side rendering in your Angular applications.
+authors:
+  - markpieszak
+tags:
+  - angular
+  - performance
+  - ssr
+  - angular-universal
+---
+
+_You can find the code samples from this article [on GitHub](https://github.com/trilonio/angular-universal-tutorial)_
+
+In this post, you'll learn how to use the [Angular Universal schematics](https://www.github.com/angular/universal) to set up dynamic server-side rendering (SSR). SSR can give you more control over your search engine optimization (SEO) and social media previews and get faster [First Meaningful Paints](/first-meaningful-paint).
+
+{% Aside %}
+
+This post assumes you're already familiar with SSR and the Angular Universal schematics and their benefits. If you need a refresher on check out the [Getting started with server-side rendering in Angular](/getting-started-with-server-side-rendering-in-angular/) post.
+
+{% endAside %}
+
+## Dynamic Server-side rendering with Angular
+
+Dynamic SSR is done at **_run time_**. With dynamic SSR, you'll need a Node.js server to handle incoming requests for routes by serializing your application to an HTML string that's then returned to the browser to be rendered.
+
+Occasionally your applications have very dynamic content or components. A common use case would be if you had routes that required SSR but had dynamic parameters as part of their url (ie: `/products/:id`).
+
+Your Node.js server will be essentially serving as a middleman for when a specific product is requested, generating the content on the fly for the given URL.
+
+To achieve this easily, you'll need to install the `@nguniversal/express-engine` your Angular CLI application. Read more about installing the schematic in the [Getting started with server-side rendering in Angular](/getting-started-with-server-side-rendering-in-angular/) post.
```

Comment:
```suggestion
To achieve this, you'll need to install the `@nguniversal/express-engine` schematic in your Angular CLI application. Read more about installing the schematic in the [Get started with server-side rendering in Angular](/getting-started-with-server-side-rendering-in-angular/) post.
```
We don't say things will be easy because it may not be for all readers, and we don't want them to have a negative emotional response.